### DISCUSSION TITLE
Is COG/ titiler is suitable for representing time series raster data?

### DISCUSSION THREAD

**Discussion by mortezaomidi** [developmentseed/titiler#543] [Q&A]
Hello,
First of all, I need to thank you for your efforts.
I have a question:
I need to display time series data of various satellite images for the purpose of drought monitoring. (for example MODIS NDVI and ...) from the year 2000 to the last day that the data exists. After preprocessing using GDAL data stored in TIFF format. (For example, each monthly data is a TIFF file)
Is titiler and generally COG are suitable for this case (representing time series data)? 
The output of my work is something similar to [wentusky - temprature product.](https://www.ventusky.com/?p=31.8;58.7;4&l=rain-3h)
Thank you for your help

**Comment by vincentsarago** [comment]
Hi @mortezaomidi, if you're goal is to quickly get values for the whole time stack, I believe you'll be better served by using Zarr to store your data or to create 1 COG with each band as a month. 

If you use one COG / month, you'll have to call the titiler endpoint for each file from you're client application, and it might be messy

**Reply by mortezaomidi** [reply]
Thanks' @vincentsarago . I am not familiar with Zarr and I don't know if it is suitable or not. 
My requirements specifically:

I have 7 different data: for example precipitation, temperature, vegetation, soil moisture and a few other indicators that are derived from the data.
The time scale of these data is monthly and their spatial scale is almost 1 km, the global data is cliped/subset on the country of Iran.
A GeoTIFF  file is created for each dataset per time. for example there almost 365 files have been produced for vegetation since 2000, which will increase over the time.
 I have two main goals: 
1- display data on the client side (user selects a dataset and available time and then data represented to the user)
2- Get the values of a pixel at all times for a data. It should be noted that for each data all tiffs stacked into a single NetCDF file. I have created this file with this intention.

With this current structure, can I use [titiler](https://github.com/developmentseed/titiler) for representation of time series data ( main goal #1). Or based on your experience this structure is wrong, and I need to make changes in the structure. 
grateful
